### ISSUE TITLE
[bug] Email notification to invalid user email breaks code execution 

### ISSUE THREAD

**Issue by pandafy** [openwisp/openwisp-notifications#171]
As reported by [Sparkxxx](https://github.com/Sparkxxx) on Gitter, if a user does not have a valid email address then email sent to that user will raise `SMTPRecipientsRefused` exception.

We need to handle this exception gracefully to make sure execution of code is not affected. 

Following is traceback shared by [Sparkxxx](https://github.com/Sparkxxx):

```
File "/usr/lib/python3.7/smtplib.py", line 881, in sendmail raise SMTPRecipientsRefused(senderrs) smtplib.SMTPRecipientsRefused: {'NOT@VALID.EMAIL': (550, b'5.1.1 NOT@VALID.EMAIL: Recipient address rejected: User unknown in local recipient table')}
```

Following code is responsible for sending email:
https://github.com/openwisp/openwisp-notifications/blob/d7f20ea41bcff1c0a437b93e98cac2d0a3bbec32/openwisp_notifications/handlers.py#L197-L213

**Comment by nemesisdesign** [comment]
Yes we should catch exceptions when sending and log it with `logger.exception('Notification email sending failed')`.
Thanks for opening the issue @pandafy!

**Comment by Saurav-Shrivastav** [comment]
I can get this done.

**Comment by harshakhmk** [comment]
I would like to work on this

**Comment by pandafy** [comment]
There is already a PR open for it #174. @RahulPalve are you still working on it?

**Comment by RahulPalve** [comment]
@pandafy Yes, just the test case for bug is remaining

**Comment by pandafy** [comment]
Some work has already been done by in #174. Due to lack of activity and tests, that PR has been closed. Before opening a new PR for this issue, read the messages we left as review on #174.

**Comment by Aryamanz29** [comment]
I give it a try and refer to #174 work already been done by the contributor.

**Comment by pandafy** [comment]
Go ahead @Aryamanz29! 👍🏼

**Comment by aagman945** [comment]
This issue should be of openwisp-utils now as this piece of code is now available in send_email.py. in openwisp-utils.

**Comment by pandafy** [comment]
Good catch @aagman945! 👏🏼 
I have opened https://github.com/openwisp/openwisp-utils/issues/278

**Comment by nemesisdesign** [comment]
@pandafy next time let's use the "move issue" feature of github which preseves the history of the discussion. I think this can be closed now.
